compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp

```diff
@@ -402,6 +403,70 @@ static bool areOpsFusable(Operation *producer, Operation *consumer,
   return true;
 }
 
+/// The logic to decide fusability (using the `hasCompatibleOuterParallelLoops`)
+/// currently works when the indexing map corresponding to result of the
+/// producer and indexing map corresponding to operand in the result are not
+/// transposed with respect to each other. To find more fusion opportunities for
+/// consumer elementwise operation, the indexing maps in the consumer can be
+/// made to "align" with the indexing map of the producer to enhance fusion.
+static bool areOpsFusableAfterInterchangeOfConsumer(
```

Comment:
Maybe rename this to something like `makeConsumerFusableViaInterchange` to signal that it modifies the operation? So it's more clear it has side-effects on the indexing maps of the op.